hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOMHAAdminPrivileges.java

```diff
@@ -0,0 +1,309 @@
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.apache.hadoop.hdds.conf.OzoneConfiguration;
+import org.apache.hadoop.hdds.utils.IOUtils;
+import org.apache.hadoop.ozone.MiniOzoneCluster;
+import org.apache.hadoop.ozone.MiniOzoneHAClusterImpl;
+import org.apache.hadoop.ozone.client.BucketArgs;
+import org.apache.hadoop.ozone.client.ObjectStore;
+import org.apache.hadoop.ozone.client.OzoneBucket;
+import org.apache.hadoop.ozone.client.OzoneClient;
+import org.apache.hadoop.ozone.client.OzoneClientFactory;
+import org.apache.hadoop.ozone.client.OzoneVolume;
+import org.apache.hadoop.ozone.client.VolumeArgs;
+import org.apache.hadoop.ozone.om.exceptions.OMException;
+import org.apache.hadoop.ozone.security.acl.IAccessAuthorizer;
+import org.apache.hadoop.ozone.security.acl.OzoneNativeAuthorizer;
+import org.apache.hadoop.security.UserGroupInformation;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+
+/**
+ * Integration test for OM HA admin privileges.
+ * Tests that admin privileges are node-specific and don't transfer 
+ * when leadership changes between OM nodes.
+ */
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class TestOMHAAdminPrivileges {
+
+  private static final String OM_SERVICE_ID = "om-service-test-admin";
+  private static final int NUM_OF_OMS = 3;
+  private static final String TEST_USER = "testuser-" + 
+      RandomStringUtils.secure().nextAlphabetic(5).toLowerCase(Locale.ROOT);
+  private static final String TEST_VOLUME = "testvol-" + 
+      RandomStringUtils.secure().nextAlphabetic(5).toLowerCase(Locale.ROOT);
+  private static final String ADMIN_VOLUME = "adminvol-" + 
+      RandomStringUtils.secure().nextAlphabetic(5).toLowerCase(Locale.ROOT);
+  private static final String TEST_BUCKET = "testbucket-" + 
+      RandomStringUtils.secure().nextAlphabetic(5).toLowerCase(Locale.ROOT);
+
+  private MiniOzoneHAClusterImpl cluster;
+  private OzoneClient client;
+  private UserGroupInformation testUserUgi;
+
+  @BeforeAll
+  public void init() throws Exception {
+    // Create test user
+    testUserUgi = UserGroupInformation.createUserForTesting(TEST_USER, new String[]{"testgroup"});
+    
+    // Set up and start the cluster
+    setupCluster();
+    
+    // Create admin volume that will be used for bucket permission testing
+    createAdminVolume();
+  }
+
+  @AfterAll
+  public void shutdown() {
+    IOUtils.closeQuietly(client);
+    if (cluster != null) {
+      cluster.shutdown();
+    }
+  }
```

Comment:
Usually BeforeAll and AfterAll should be marked as static. However, I'm OK to leave it be.